tests/rbd_mirror/test_rbd_mirror_snapshot_metrics.py

```diff
@@ -0,0 +1,158 @@
+        4. Enable image mode snapshot based mirroring on the pool respectively.
+        5. Start running IOs on the primary image.
+        6. verify snapshot based performnace counter metrics.
+"""
+import os
+import re
+
+from tests.rbd.rbd_utils import Rbd
+from tests.rbd_mirror import rbd_mirror_utils as rbdmirror
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def validate_snap_metrics(mirror_node):
+    """Method to validate snapshot based mirroring metrics
+
+    Args:
+        mirror_node: rbd-mirror daemon node
+    """
+    # Change directory to the base path
+    os.chdir("/var/run/ceph")
+
+    # List the contents of the current directory
+    directory_contents = os.listdir()
+
+    # Check if there are any directories in the current directory
+    if directory_contents:
+        target_directory = directory_contents[0]
+        os.chdir(target_directory)
+        directory_contents = os.listdir()
+
+        # Find the file with the specific pattern
+        file_pattern = "ceph-client.rbd-mirror.*"
+        for file in directory_contents:
+            if re.match(file_pattern, file):
+                target_file = file
+                break
+
+    # Check if the target file exists
+    if target_file:
+        # Create a symbolic link named 'symlink_name' pointing to the target file
+        os.symlink(target_file, "rbd-mirror.asok")
+        directory_contents = os.listdir()
+
+        # Check if the symlink file named rbd-mirror.asok is in the directory_contents
+        if "rbd-mirror.asok" in directory_contents:
+            log.info("symlink file named rbd-mirror.asok created successfully")
+        else:
+            log.error("symlink file named rbd-mirror.asok not created successfully")
+            return 1
+
+    # Validation of mirror related metrics
+    command = "ceph --admin-daemon rbd-mirror.asok counter dump"
+
+    out = mirror_node.exec_cmd(cmd=command, output=True)
+
+    # Check if the required snapshot based metrics are present
+    if (
+        "rbd_mirror_snapshot" in out
+        and "labels" in out["rbd_mirror_snapshot"]
+        and "counters" in out["rbd_mirror_snapshot"]
+    ):
+        counters = out["rbd_mirror_snapshot"]["counters"]
+        if (
+            "snapshots" in counters
+            and "sync_time" in counters
+            and "sync_bytes" in counters
+        ):
+            log.info("All required snapshot based metrics are present.")
+            return 0
+    else:
+        log.error("One or more required snapshot based metrics are missing.")
+        return 1
+
+
+def run(ceph_cluster, **kw):
+    """Verification of mirroring failure,
+    when secondary cluster storage is full.
+
+    Args:
+        ceph_cluster: The storage cluster participating in the test
+        **kw: test data
+
+    Returns:
+        0 - if test case pass
+        1 - if test case fails
+    """
+
+    try:
+        log.info(
+            "Starting CEPH-9507 - Verification of mirroring failure "
```

Comment:
Change this to the test being automated.

Reply:
Done